### PR TITLE
Update preferences

### PR DESCRIPTION
**Pull request by domdomegg**
Use `PreferenceFragment`, easier to maintain preferences, recommended way of doing it in API level > 11 (our minSdkVersion is 15). Also have updated SingleUploadFragment and Utils to work better, and use new arrays.xml.

I think the most controversial change is requiring Utils helper methods to be passed a context. This is to access the strings so we don't duplicate stuff e.g. creating `Licenses` in `Prefs.java` - there's already so many similar string names, we don't need more!

As this is quite a big change affecting a large number of files I think it's best if it is approved by at least 2 people before it's merged.

A quick explanation of what I've done:
- Removed `Licenses` constant in Prefs, prefer using `arrays.xml` (which in turn inherits from `strings.xml`)
- Move most of the code in SettingsActivity into SettingsFragment, and remove code no longer needed to force an ActionBar
- setOnPreferenceChangeListener's on the spinner to update the summary as before, and recreate the activity to update the theme straight away when changed
- Tidy up Utils, remove code blocks for API levels < 11 (current minSdkVersion is 15)
- Require context to be passed to Utils to compare strings
- Update SingleUploadFragment to not duplicate values again, use `arrays.xml` instead (which in turn inherits from `strings.xml`). May be an idea to move large if block getting license shorthand from name from SingleUploadFragment to Utils.
- Move "Become a beta tester" link in settings to bottom

Please do ask any questions about anything else, will either help spot potential bugs or you'll be able to maintain it easier later :)

### CODE REVIEW

**Comment by misaochan** [comment]
@domdomegg , have you tested that everything runs with at least API 21, 23 and 25? I'm a bit wary of the Utils change because there are many, many usages throughout the app, so high chance of stuff breaking. But if it has been thoroughly tested then the rationale behind it sounds good to me.

**Comment by domdomegg** [comment]
Tested with API 21, 23, 24 (physical device), 25 and 15 - found a bug in 15 that seemed to turn up in all the other ones only after I ran it through API 15 - more likely me just being crazy.

New commit should fix said bug, and it works on levels above (although with preferences its sometimes dodgy as to whether it saves it when reinstalling/uninstalling - but recreating 25 and 23 devices got same results in test).

Probably still worth someone else doing a `checkout` and testing it on one API level - should work first time, otherwise please do post a stacktrace.

The main things that probably need testing are:
- Opening settings, messing around with them for a while - especially the license ListPreference
- Logging in from a clean install (as this is where default preferences are set)
- The license spinner on upload

**Comment by domdomegg** [comment]
@codacy-bot 
image

But seriously, I think that's the best I'm going to get it for Codacy - just line lengths where I think it would be impractical to split a line and a Javadoc for a function I didn't really alter. Should be good to merge after someone tests it.

**Comment by veyndan** [comment]
It looks like you are changing a lot in one PR. Personally I think it'd be better to split them into multiple PRs for two reasons:

1. Ease of testing the PR.
2. If we need to revert any changes and this PR is seen to be the problem, then we are reverting a lot of changes whereas if it was split into multiple PRs we would only need to revert the PR that caused the problem, hence isolating the issue would be a lot easier.

A lot of the bullet points could be made into one or two PRs themselves.

**Comment by domdomegg** [comment]
I think it looks like I'm changing a lot but that's probably just because of the stupid number of commits I made messing around with stuff (should probably squash and merge this one to avoid cluttering commit messages).

I mean potentially you could have 3 PRs:
- Split PreferenceFragment and PreferenceActivity
- Use `arrays.xml` in `preferences.xml`
- Update Utils and SingleUploadActivity to use `arrays.xml`
But they are all quite linked, especially 2 & 3 there and trying to separate them would be a load of extra work rather than just updating them all at the same time.

In regards to your second advantage, because of how intertwined they are I doubt it would be easy to revert just the PR that caused the problem anyways as you'd end up with a load of merge conflicts everywhere.

I can split them up into various PRs if you wish. I'll take on board that I should probably make smaller PRs in the future though either way :)

**Comment by veyndan** [comment]
Yeah it'd be good if you could squash them, it would be easier to understand the intent of the commits. As it is intertwined I wouldn't bother for now but in the future that'd be good. However, I'm not sure if I'll have time to review the PR so I guess it's up to the discretion of whoever reviews it.

Also I'd suggest rebasing as 2686a51 was done in #453.

**Comment by misaochan** [comment]
I'll try to test and review later this weekend, and I agree that a 2nd reviewer would be a good idea for this PR. Sorry for the long wait @domdomegg , but it is truly massive. :)

**Comment by domdomegg** [comment]
No worries - thank you for reviewing it!

Will have a look at `menu/fragment_contributions_list.xml` when I get home

File: app/src/main/res/menu/fragment_contributions_list.xml [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
+        android:title="@string/menu_about"
+        app:showAsAction="never"
         />
     <item android:id="@+id/menu_settings"
-          android:title="@string/menu_settings"
-          app:showAsAction="never"
+        android:title="@string/menu_settings"
+        app:showAsAction="never"
+        />
+    <item android:id="@+id/menu_feedback"
```
Comment (whym) [inline app/src/main/res/menu/fragment_contributions_list.xml]: Is this an unnecessary duplicate of menu_feedback?

**Comment by misaochan** [comment]
I wonder if it would be possible to extract the Utils.java changes into a separate PR at this stage? Those are the ones that I am most wary of due to extensive usage throughout the app. The other changes look okay to me so far and I would be happy to merge them, but having the Utils.java ones separated would help a lot. 

Not sure if this is doable at this stage though?

**Comment by domdomegg** [comment]
I'm going to split this up later and do it without the horrible commit history mess that we have here.

**Comment by misaochan** [comment]
Thanks @domdomegg !

**Pull request closed without merging**